cases-SEP24/transactions.test.js

```diff
@@ -145,6 +145,39 @@ describe("Transactions", () => {
     expect(json.transactions.length, logs).toBe(1);
   });
 
+  it("returns a transactions in decending order", async () => {
+    await createTransaction({
+      currency: enabledCurrency,
+      account: keyPair.publicKey(),
+      toml: toml,
+      jwt: jwt,
+      isDeposit: true,
+    });
+    await createTransaction({
+      currency: enabledCurrency,
+      account: keyPair.publicKey(),
+      toml: toml,
+      jwt: jwt,
+      isDeposit: true,
+    });
+
+    const { json, status, logs } = await loggableFetch(
+      transferServer + `/transactions?asset_code=${enabledCurrency}`,
+      {
+        headers: {
+          Authorization: `Bearer ${jwt}`,
+        },
+      },
+    );
+    let expected = json.transactions.sort((a, b) =>
+      a.started_at > b.started_at ? 1 : -1,
```

Comment:
> If compareFunction(a, b) returns less than 0, sort a to an index lower than b (i.e. a comes first).
> If compareFunction(a, b) returns greater than 0, sort b to an index lower than a (i.e. b comes first).

Reading this from the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) it seems we should return `-1` when `a` is greater than `b` right?

Reply:
I would also check to see how `.sort()` handles `null` and `undefined` values. `started_at` is a required field so we should fail if objects in `json.transactions` have `null` or `undefined` values, or if its missing the attribute entirely.

Reply:
`Reading this from the docs it seems we should return -1 when a is greater than b right?`
You're right! I'll fix that now.

Reply:
Use the '>' character to quote :-)

Reply:
```
I would also check to see how .sort() handles null and undefined values. started_at is a required field so we should fail if objects in json.transactions have null or undefined values, or if its missing the attribute entirely.
```
If the transaction's `started_at` key is null then its sorted to be at the end of the list.
If `started_at` is missing from one of the transactions it'll be in the front of the array.

I'm unsure if either of these would be desired behavior.
Should I have an initial check to ensure that the json returned is the correct schema using 
`getTransactionSchema`?
Or is that out of scope for the test?

Reply:
Preferably we do a linear check for bad `started_at` values. I wouldn't want to fail on an unrelated schema issue.

Reply:
Just to be clear "bad" `started_at` values are anything not a `UTC ISO 8601 string`.
Like the example you gave `started_at: null`.
If thats correct I'll implement a linear check to ensure the transactions have `UTC ISO 8601 string` timestamps for their `started_at` values.